### PR TITLE
Remove deprecated packaging types

### PR DESCRIPTION
**Pull request by mickaelistria**

### CODE REVIEW

File: tycho-p2/tycho-p2-facade/src/main/java/org/eclipse/tycho/p2/resolver/P2DependencyResolver.java [hunk 1]
```diff
@@ -449,7 +449,7 @@ private DependencyArtifacts doResolveDependencies(MavenSession session, MavenPro
     private boolean isAllowConflictingDependencies(MavenProject project, TargetPlatformConfiguration configuration) {
         String packaging = project.getPackaging();
 
-        if (PackagingType.TYPE_ECLIPSE_UPDATE_SITE.equals(packaging)
+        if (PackagingType.TYPE_ECLIPSE_REPOSITORY.equals(packaging)
```
Comment (laeubi) [inline tycho-p2/tycho-p2-facade/src/main/java/org/eclipse/tycho/p2/resolver/P2DependencyResolver.java:452]: I wonder if this should not be removed here as the `TYPE_ECLIPSE_REPOSITORY` was previously not included here?

File: tycho-release/tycho-versions-plugin/src/main/java/org/eclipse/tycho/versions/manipulation/CategoryXmlManipulator.java [hunk 2]
```diff
@@ -126,4 +125,11 @@ public void writeMetadata(ProjectMetadata project) throws IOException {
         // this manipulator does not add any restrictions on version changes allowed for eclipse-repository projects
         return null;
     }
+
+    private static String rewriteFeatureUrl(String url, VersionChange change) {
```
Comment (laeubi) [inline tycho-release/tycho-versions-plugin/src/main/java/org/eclipse/tycho/versions/manipulation/CategoryXmlManipulator.java:129]: Is this related?

Comment (mickaelistria) [reply tycho-release/tycho-versions-plugin/src/main/java/org/eclipse/tycho/versions/manipulation/CategoryXmlManipulator.java:129]: Not really, it could be extracted into a separate patch

**Comment by akurtakov** [comment]
This PR tries to achieve too much at once thus I'll be trying to handle things separately so I can actually merge them rather then constantly add to the pile. E.g. #865

**Comment by laeubi** [comment]
> This PR tries to achieve too much at once thus I'll be trying to handle things separately so I can actually merge them rather then constantly add to the pile. E.g. #865

Good idea, I have just boot up the build-node to speed up PR build queue, should we then close this one for now?

**Comment by akurtakov** [comment]
Yeah, let's close it.
